### PR TITLE
[stdlib] Implement List.__repr__() method

### PR DESCRIPTION
**Pull request by msaelices**
Based on the `List.__str__()` one, to allow to call `repr(some_list)`.

It's not currently passing because of the following error, but I think the code is correct and it's a limitation of the mojo compiler, which does not understand that, if `self` is made of `RepresentableCollectionElement`, the `__repr__` method can call the `__str__`  one, as `__type_of(self)` will also be `List[RepresentableCollectionElement]` 

image

### CODE REVIEW

**Review by gabrieldemarmiesse — COMMENTED** [review]
Thank you for the pull request! If you remove the trait, this PR should be good to merge :)

File: stdlib/test/builtin/test_list.mojo [hunk 1]
```diff
@@ -31,6 +31,12 @@ fn test_variadic_list() raises:
     check_list(5, 8, 6)
 
 
+fn test_repr_list() raises:
```
[the remark this answers is not in the document]
Comment (msaelices) [reply stdlib/test/builtin/test_list.mojo:34]: Done: https://github.com/msaelices/mojo/commit/f552e501235d1f8f75c403861ec5d7ff38449bd2

File: stdlib/src/collections/list.mojo [hunk 2]
```diff
@@ -72,7 +72,9 @@ struct _ListIter[
             return self.index
 
 
-struct List[T: CollectionElement](CollectionElement, Sized, Boolable):
+struct List[T: CollectionElement](
+    CollectionElement, Sized, Boolable, Representable
```
[the remark this answers is not in the document]
Comment (msaelices) [reply stdlib/src/collections/list.mojo]: Done: https://github.com/msaelices/mojo/commit/1c1dec97782084942c9093b378484e4a7fa686bb

**Review by gabrieldemarmiesse — APPROVED** [review]
Looks good! I think it's ready to be merged. Thank you for your contribution!

**Review by JoeLoser — APPROVED** [review]
Looks good, thank you! We have a trick proven out today for the `List.index()` method that I think we can apply here as well when the new nightly is available later tonight/tomorrow.  I'll merge this internally in the meantime and we can refactor it as a static method to avoid the `__type_of` dance.

**Pull request closed without merging**